odoo/orm/environments.py

```diff
@@ -837,7 +851,7 @@ def get_value(id_):
                 break
         return vals
 
-    def get_records_different_from(self, records, field, value):
+    def get_records_different_from(self, records: BaseModel, field: Field, value: typing.Any) -> BaseModel:
```

Comment:
This could be more precise.  Introduce `M = typing.TypeVar("M", bound=BaseModel)` for model classes, and you can infer that the output is of the same subclass as the input:
```suggestion
    def get_records_different_from(self, records: M, field: Field, value: typing.Any) -> M:
```